google_guest_agent/main.go

```diff
@@ -79,45 +76,41 @@ func logStatus(name string, disabled bool) {
 	logger.Infof("GCE %s manager status: %s", name, status)
 }
 
-func parseConfig(file string) (*ini.File, error) {
-	// Priority: file.cfg, file.cfg.distro, file.cfg.template
-	cfg, err := ini.LoadSources(ini.LoadOptions{Loose: true, Insensitive: true}, file, file+".distro", file+".template")
-	if err != nil {
-		return nil, err
-	}
-	return cfg, nil
-}
-
 func closeFile(c io.Closer) {
 	err := c.Close()
 	if err != nil {
 		logger.Warningf("Error closing file: %v.", err)
 	}
 }
 
-func runUpdate(ctx context.Context) {
+func runUpdate(ctx context.Context, config *cfg.Sections) {
 	var wg sync.WaitGroup
 	mgrs := []manager{&addressMgr{}}
 	switch runtime.GOOS {
 	case "windows":
-		mgrs = append(mgrs, []manager{newWsfcManager(), &winAccountsMgr{}, &diagnosticsMgr{}}...)
+		mgrs = append(mgrs, []manager{newWsfcManager(config), &winAccountsMgr{}, &diagnosticsMgr{}}...)
 	default:
 		mgrs = append(mgrs, []manager{&clockskewMgr{}, &osloginMgr{}, &accountsMgr{}}...)
 	}
 	for _, mgr := range mgrs {
 		wg.Add(1)
 		go func(mgr manager) {
 			defer wg.Done()
-			if mgr.disabled(runtime.GOOS) {
+			if disabled, _ := mgr.Disabled(ctx, config); disabled {
```

Comment:
I understand error is not used right now and all mgrs are returning nil, but it would be good to add error check right now so we don't have to remember about it when implementation actually starts throwing errors.

Reply:
done. thanks.